internal/eval/eval.go

```diff
@@ -675,6 +675,162 @@ func evalJSONSET(args []string, store *dstore.Store) []byte {
 	return clientio.RespOK
 }
 
+// evalJSONARRAPPEND appends the value(s) provided in the args to the given array path
+// in the JSON object saved at key in arguments.
+// Args must contain atleast a key, path and value.
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the object at given path is not an array, it returns response.RespNIL.
+// Returns the new length of the array at path.
+func evalJSONARRAPPEND(args []string, store *dstore.Store) []byte {
+    if len(args) < 3 {
+        return diceerrors.NewErrArity("JSON.ARRAPPEND")
+    }
+
+    var key string = args[0]
+    var path string = args[1]
+
+    obj := store.Get(key)
+    if obj == nil {
+        return diceerrors.NewErrWithMessage("could not perform this operation on a key that doesn't exist")
+    }
+
+    errWithMessage := dstore.AssertTypeAndEncoding(obj.TypeEncoding, dstore.ObjTypeJSON, dstore.ObjEncodingJSON)
+    if errWithMessage != nil {
+        return errWithMessage
+    }
+
+    jsonData := obj.Value
+    _, err := sonic.Marshal(jsonData)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("Existing key has wrong Dice type")
+    }
+
+    expr, err := jp.ParseString(path)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("invalid JSONPath")
```

Comment:
The error message should be something like `ERR Path '<path>' does not exist or not an array`